src/jmh/java/net/sf/jabref/benchmarks/Benchmarks.java

```diff
@@ -0,0 +1,91 @@
+package net.sf.jabref.benchmarks;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import net.sf.jabref.*;
+import net.sf.jabref.exporter.BibDatabaseWriter;
+import net.sf.jabref.exporter.SaveException;
+import net.sf.jabref.exporter.SavePreferences;
+import net.sf.jabref.importer.ParserResult;
+import net.sf.jabref.importer.fileformat.BibtexParser;
+import net.sf.jabref.logic.search.SearchQuery;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.database.BibDatabaseMode;
+import net.sf.jabref.model.database.BibDatabaseModeDetection;
+import net.sf.jabref.model.entry.BibEntry;
+import org.openjdk.jmh.Main;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.runner.RunnerException;
+
+@State(Scope.Thread)
+public class Benchmarks {
+    StringReader bibtexStringReader;
+    BibDatabase database = new BibDatabase();
+
+    @Setup
+    public void init() throws IOException, SaveException {
+        Globals.prefs = JabRefPreferences.getInstance();
+
+        Random randomizer = new Random();
+        for(int i = 0; i < 1000; i++)
+        {
```

Comment:
formatting

I would use more elements, e.g., 10k as this should be more helpful for finding the issues.